libcxx/include/__atomic/atomic.h

```diff
@@ -74,7 +74,7 @@ struct atomic<_Tp*> : public __atomic_base<_Tp*> {
   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR atomic(_Tp* __d) _NOEXCEPT : __base(__d) {}
 
   _LIBCPP_HIDE_FROM_ABI _Tp* operator=(_Tp* __d) volatile _NOEXCEPT
-    requires __base::is_always_lock_free
+    _LIBCPP_REQUIRE_IS_ALWAYS_LOCK_FREE
```

Comment:
I'm not sure how to test these changes to `std::atomic<T*>` since pointer type since virtually all systems with atomic instructions will support atomic reads and writes of pointer width. 

Perhaps I should constrain this to architectures where this is not the case?

Reply:
> since virtually all systems with atomic instructions will support atomic reads and writes of pointer width

This is not guaranteed/required by the standard.

I think the approach in microsoft/STL#634 works - define a conditionally deprecated but always `true` constant and then `static_assert` it. With this approach you don't need to be worried about whether some atomic types are always lock-free.

Reply:
I understand that it is not a guarantee, but I am not sure I an write a test that will run on any ISAs i have access to since ARM and x86 both have instructions to atomically write 16 bytes of data.

Perhaps I can restrict the test to MIPS or some simpler architecture? I'm not familiar enough with the testing infrastructure to know the correct manner to proceed here.

Reply:
> I think the approach in [microsoft/STL#634](https://github.com/microsoft/STL/pull/634) works - define a conditionally deprecated but always `true` constant and then `static_assert` it. With this approach you don't need to be worried about whether some atomic types are always lock-free.

I am pursuing this right now, as it seems to be the only way to consistently raise deprecated warnings.